sunpy/io/tests/test_asdf.py

```diff
@@ -0,0 +1,33 @@
+import numpy as np
+
+import asdf
+
+from sunpy.data.test import get_test_filepath
+from sunpy.io._asdf import get_header, get_keys_name, read, write
+from sunpy.io._header import FileHeader
+
+AIA_MAP = get_test_filepath("aiamap_genericmap_1.0.0.asdf")
```

Comment:
Did you create this yourself? I thought we already have asdf test files that can be used for these tests.

Reply:
I ran this code in data/test/generate_asdf_test.py  and I used the file generated by this code 
https://github.com/sunpy/sunpy/blob/main/sunpy/data/test/_generate_asdf_test.py

Reply:
We have existing asdf test files, so was there a need to create a new one and not use those?

Reply:
i thought the one that existed before might be an older version since after running the _generate_test_asdf.py gave new file which had no shift defined. The previous one had a map with shift

Reply:
Does the shift matter?

Reply:
well with or without shift the read file should return same thing

Reply:
So we can use the existing files instead for the test or will they not work?

Reply:
yes it should work

Reply:
Sure.